mirascope/chat/tools.py

```diff
@@ -42,3 +49,38 @@ def tool_schema(cls) -> ChatCompletionToolParam:
             }
 
         return cast(ChatCompletionToolParam, {"type": "function", "function": fn})
+
+    @classmethod
+    def from_tool_call(cls, tool_call: ChatCompletionMessageToolCall) -> OpenAITool:
+        """Returns an instance of the tool constructed from a tool call response."""
+        return cls(**json.loads(tool_call.function.arguments))
+
+
+T = TypeVar("T", bound=OpenAITool)
+
+
+def openai_tool_fn(fn: Callable) -> Callable:
+    """A decorator for adding a function to a tool class.
+
+    Adding this decorator will add an `fn` property to the tool class that returns the
+    function that the tool describes. This is convenient for calling the function given
+    an instance of the tool.
+
+    Args:
+        fn: The function to add to the tool class.
+
+    Returns:
+        The decorated tool class.
+    """
+
+    def decorator(cls: Type[T]) -> Type[T]:
+        """A decorator for adding a function to a tool class."""
+
+        def fn_property(self) -> Callable:
+            """Returns the function that the tool describes."""
+            return fn
+
+        setattr(cls, "fn", property(fn_property))
```

Comment:
do you need to define fn_property. could you pass fn to setattr instead?

Reply:
You need to define self, but we can further simplify using a lambda